test/superfluid.test.ts

```diff
@@ -319,6 +323,104 @@ describe('Superfluid', () => {
         ).to.be.revertedWith('SUPERFLUID_INVALID_SUPERTOKEN');
       });
     });
+
+    describe('when upgrading Super Token amount', () => {
+      const amountToUpgrade = toDecimals(200);
+
+      beforeEach(async () => {
+        await mintFakeTokens(ffAgentSigner, amountToUpgrade);
+      });
+
+      it('should upgrade Token amount on agent correctly', async () => {
+        const agentFakeTokenBalanceBefore = await ffAgent.balance(fakeToken.address);
+
+        await superfluidApp.upgrade(superToken.address, amountToUpgrade);
+
+        const agentFakeTokenBalanceAfter = await ffAgent.balance(fakeToken.address);
+
+        expect(agentFakeTokenBalanceBefore.sub(amountToUpgrade)).equal(agentFakeTokenBalanceAfter);
+      });
+
+      it('should upgrade Super Token amount on agent correctly', async () => {
+        const agentSuperTokenBalanceBefore = await ffAgent.balance(superToken.address);
+
+        await superfluidApp.upgrade(superToken.address, amountToUpgrade);
+
+        const agentSuperTokenBalanceAfter = await ffAgent.balance(superToken.address);
+
+        expect(agentSuperTokenBalanceBefore.add(amountToUpgrade)).equal(
+          agentSuperTokenBalanceAfter
+        );
+      });
+
+      it('should revert when trying to upgrade Super Token without having MANAGE_STREAMS_ROLE', async () => {
+        await expect(
+          superfluidApp.connect(permissionlessAccount).upgrade(superToken.address, 0)
+        ).to.be.revertedWith('APP_AUTH_FAILED');
+      });
+
+      it('should revert if provided token is not SuperToken', async () => {
+        await expect(superfluidApp.upgrade(fakeToken.address, amountToUpgrade)).to.be.revertedWith(
+          'SUPERFLUID_INVALID_SUPERTOKEN'
+        );
+      });
+
+      it('should revert when trying to upgrade Super Token with amount less then 1', async () => {
+        await expect(superfluidApp.upgrade(superToken.address, 0)).to.be.revertedWith(
+          'SUPERFLUID_UPGRADE_AMOUNT_ZERO'
+        );
+      });
+    });
+
+    describe('when downgrading Super Token amount', () => {
+      const amountToDowngrade = toDecimals(200);
+
+      beforeEach(async () => {
+        await mintFakeTokens(ffAgentSigner, amountToDowngrade);
+      });
```

Comment:
I think we can remove this `beforeEach` because we do not need to have `FakeTokens` in the Agent app. And we already minted SuperTokens during the `Prepare agent app` before step.

Reply:
I got you, sure, on downgrade we do not need to mint new Fake Tokens